src/i18n/Data/Intl/IntlLocales.php

```diff
@@ -1605,9 +1605,11 @@ public function validate($locale)
         if (!$lang || !$region) {
             return false;
         }
-        return strcasecmp($lang, $region)
```

Comment:
Removing the $lang and $region comparison should be all you need to do to fix this issue.